internal/chart/generate.go

```diff
@@ -0,0 +1,47 @@
+package chart
+
+import (
+	"errors"
+	"fmt"
+)
+
+var (
+	ErrMalformedConfiguration = fmt.Errorf("logic error, malformed configuration generated")
+	ErrGenerationFailed       = fmt.Errorf("failed to generate configuration")
+)
+
+type (
+	configVisitor = func(configuration *Configuration) (yamlMap, error)
+	yamlGenerator struct {
+		visitors map[string]configVisitor
+	}
+)
+
+func (g *yamlGenerator) AddVisitor(name string, visitor configVisitor) {
```

Comment:
What is visitor? It's not clear for me what's he is doing

Reply:
I'd say it's visitor pattern which partially conforms to generally used rules. It's basically an entity which visits configuration without changing that and generates some YAML configuration based on some options. 

I decided to group them into separate entities because it'll become a pain to manage if we'll have "all-in-one" generator